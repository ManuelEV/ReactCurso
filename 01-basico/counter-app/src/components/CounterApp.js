import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(0);

    const onAdd = () => {
        setCounter(counter + 1)
        // setCounter( (c) => c + 1 );
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ onAdd }>ADD</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

export default CounterApp;