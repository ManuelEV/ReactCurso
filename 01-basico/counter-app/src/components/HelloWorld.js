import React from 'react';
import PropTypes from 'prop-types';
// Functional component
const HelloWorld = ({ myProp, myDefaultProp }) => {
    const testVar = 'XD';
    const object = {
        name: 'Juan',
        age: 20,
    };
    return (
        <>
            <h1>Hello World</h1>
            <h1>{ myProp }</h1>
            <h2>{ myDefaultProp }</h2>
            <p>{ testVar }</p>
            <pre>{JSON.stringify(object, null, 3)}</pre>
        </>
    );
}

HelloWorld.propTypes = {
    myProp: PropTypes.string.isRequired,
}

HelloWorld.defaultProps = {
    myDefaultProp: 'IM A DEFAULT PROP',
}

export default HelloWorld;