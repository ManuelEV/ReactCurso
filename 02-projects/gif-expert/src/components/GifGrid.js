import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const baseUrl = 'https://api.giphy.com/v1/gifs/search?limit=10';
        const query = category;
        const apiKey = 'N2klpHspRpcd7vphx0sGYSeBlOd60YyH';
        const url = baseUrl + '&q=' + query + '&api_key=' + apiKey;

        const response = await fetch(url);

        const {data} = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        setImages(gifs);
    }

    getGifs();

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                    {
                        images.map((img) => (
                            <GifGridItem 
                                key = {img.id}
                                {...img}
                            />
                        ))
                    }
            </div>
        </>
    )
}
