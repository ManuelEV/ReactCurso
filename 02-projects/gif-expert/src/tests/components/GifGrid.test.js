import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente GifGid', () => {

    const category = 'Rick and Morty'
    

    test('debe de coincidir el SnapShot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [
            {
                id: 'abc',
                url: 'http://localhost',
                title: 'Cualquier cosa'
            },
            {
                id: '123',
                url: 'http://localhost',
                title: 'Cualquier cosa 2'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.find('p').exists()).toBe(false);

        // revisa que la cantidad de gifs sea igual a las veces que se renderiza GifGridItem
        
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })

})

