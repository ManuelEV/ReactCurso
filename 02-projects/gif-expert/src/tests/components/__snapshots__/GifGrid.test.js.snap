// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGid debe de coincidir el SnapShot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente GifGid debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="http://localhost"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa 2"
      url="http://localhost"
    />
  </div>
</Fragment>
`;
